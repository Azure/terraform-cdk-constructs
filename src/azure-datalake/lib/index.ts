export * from "./filesystem";
export * from "./path";
export * from "./datalake";
