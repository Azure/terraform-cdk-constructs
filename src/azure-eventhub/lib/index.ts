export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
