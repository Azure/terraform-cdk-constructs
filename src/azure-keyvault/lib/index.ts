export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

