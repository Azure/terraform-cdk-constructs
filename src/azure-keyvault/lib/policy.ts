import { KeyVaultAccessPolicyA } from "@cdktf/provider-azurerm/lib/key-vault-access-policy";
import { Construct } from "constructs";
import { Vault } from "./vault";

export interface AccessPolicyProps {
  /**
   * The Azure Key Vault instance or its identifier.
   */
  readonly keyVaultId: Vault;

  /**
   * The Azure Active Directory tenant ID where the Key Vault is hosted.
   * This is typically the directory ID of your Azure AD.
   */
  readonly tenantId: string;

  /**
   * The Azure Active Directory object ID for which the policy will be applied.
   * This can be a user, group, or service principal.
   */
  readonly objectId: string;

  /**
   * The permissions to secrets stored in the Key Vault.
   * Possible values might include: 'get', 'list', 'set', 'delete', etc.
   * If not provided, no secret permissions are set.
   */
  readonly secretPermissions?: string[];

  /**
   * The permissions to certificates stored in the Key Vault.
   * Possible values might include: 'get', 'list', 'create', 'update', etc.
   * If not provided, no certificate permissions are set.
   */
  readonly certificatePermissions?: string[];

  /**
   * The permissions to keys stored in the Key Vault.
   * Possible values might include: 'get', 'list', 'create', 'sign', etc.
   * If not provided, no key permissions are set.
   */
  readonly keyPermissions?: string[];

  /**
   * The permissions to storage accounts linked to the Key Vault.
   * Possible values might include: 'get', 'list', 'delete', 'set', etc.
   * If not provided, no storage permissions are set.
   */
  readonly storagePermissions?: string[];
}

export class AccessPolicy extends Construct {
  public readonly fqdn: string;

  /**
   * Constructs a new Access Policy for Azure Key Vault.
   *
   * This class is responsible for setting up access policies that define what operations an Azure AD identity
   * can perform on the keys, secrets, certificates, and storage accounts within a specified Azure Key Vault.
   *
   * @param scope - The scope in which to define this construct, usually representing the Cloud Development Kit (CDK) stack.
   * @param id - The unique identifier for this instance of the access policy.
   * @param props - The properties for creating the access policy as defined in AccessPolicyProps. These include:
   *                - `keyVaultId`: The Azure Key Vault identifier where the policy will be set.
   *                - `tenantId`: The tenant ID of the Azure AD tenant where the Key Vault is hosted.
   *                - `objectId`: The object ID of the Azure AD identity (user, group, or service principal).
   *                - `secretPermissions`: Optional list of permissions to secrets within the Key Vault.
   *                - `certificatePermissions`: Optional list of permissions to certificates within the Key Vault.
   *                - `keyPermissions`: Optional list of permissions to keys within the Key Vault.
   *                - `storagePermissions`: Optional list of permissions to storage accounts linked to the Key Vault.
   *
   * Example usage:
   * ```typescript
   * new AccessPolicy(this, 'MyAccessPolicy', {
   *   keyVaultId: myKeyVault,
   *   tenantId: 'my-tenant-id',
   *   objectId: 'user-object-id',
   *   keyPermissions: ['get', 'list', 'update'],
   *   secretPermissions: ['get'],
   *   certificatePermissions: ['get', 'list'],
   *   storagePermissions: ['get', 'list']
   * });
   * ```
   */
  constructor(scope: Construct, id: string, props: AccessPolicyProps) {
    super(scope, id);

    const policy = new KeyVaultAccessPolicyA(this, "policy", {
      keyVaultId: props.keyVaultId.id,
      tenantId: props.tenantId,
      objectId: props.objectId,
      secretPermissions: props.secretPermissions,
      certificatePermissions: props.certificatePermissions,
      keyPermissions: props.keyPermissions,
      storagePermissions: props.storagePermissions,
    });

    this.fqdn = "azurerm_key_vault_access_policy." + policy.friendlyUniqueId;
  }
}
