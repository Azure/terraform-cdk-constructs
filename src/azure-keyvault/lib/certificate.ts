import { KeyVaultCertificate } from "@cdktf/provider-azurerm/lib/key-vault-certificate"; // Adjust the import path based on the actual module location.
import { KeyVaultCertificateIssuer } from "@cdktf/provider-azurerm/lib/key-vault-certificate-issuer";
import { Construct } from "constructs";
import { AccessPolicy } from "./policy";
import { Vault } from "./vault";

export interface SelfSignedCertificateProps {
  /**
   * The name of the certificate in the Azure Key Vault.
   */
  readonly name: string;

  readonly subject: string;

  readonly dnsNames: string[];

  readonly keyVaultId: Vault;

  readonly actionType?: string;

  readonly daysBeforeExpiry?: number;

  readonly accessPolicies: AccessPolicy[];

  readonly tags?: { [key: string]: string };
}

export interface CertificateIssuerProps {
  /**
   * The name of the certificate issuer in the Azure Key Vault.
   */
  readonly name: string;

  readonly providerName: string;

  readonly keyVaultId: Vault;

  readonly accessPolicies: AccessPolicy[];

  readonly username?: string;

  readonly password?: string;
}

export class SelfSignedCertificate extends Construct {
  public certificate: KeyVaultCertificate;

  constructor(scope: Construct, id: string, props: SelfSignedCertificateProps) {
    super(scope, id);

    // Provide default values
    const actionType = props.actionType ?? "AutoRenew";
    const daysBeforeExpiry = props.daysBeforeExpiry ?? 45;

    const certificate = new KeyVaultCertificate(
      this,
      "AzureKeyVaultCertificate",
      {
        name: props.name,
        keyVaultId: props.keyVaultId.id,
        certificatePolicy: {
          issuerParameters: {
            name: "Self",
          },
          keyProperties: {
            keyType: "RSA",
            keySize: 2048,
            reuseKey: true,
            exportable: true,
          },
          lifetimeAction: [
            {
              action: {
                actionType: actionType,
              },
              trigger: {
                daysBeforeExpiry: daysBeforeExpiry,
              },
            },
          ],
          secretProperties: {
            contentType: "application/x-pkcs12",
          },
          x509CertificateProperties: {
            subject: props.subject,
            subjectAlternativeNames: {
              dnsNames: props.dnsNames,
            },
            keyUsage: [
              "cRLSign",
              "dataEncipherment",
              "digitalSignature",
              "keyAgreement",
              "keyCertSign",
              "keyEncipherment",
            ],
            extendedKeyUsage: ["1.3.6.1.5.5.7.3.1"],
            validityInMonths: 12,
          },
        },
      },
    );
    this.certificate = certificate;

    // Accumulate all the fqdn values
    const dependencies: string[] = [];
    for (const policy of props.accessPolicies) {
      dependencies.push(policy.fqdn);
    }

    // Add dependency on all access policies
    certificate.addOverride("depends_on", dependencies);
  }
}

export class CertificateIssuer extends Construct {
  constructor(scope: Construct, id: string, props: CertificateIssuerProps) {
    super(scope, id);

    const certificateIssuer = new KeyVaultCertificateIssuer(
      this,
      "AzureKeyVaultCertificate",
      {
        name: props.name,
        keyVaultId: props.keyVaultId.id,
        providerName: props.providerName,
      },
    );

    // Accumulate all the fqdn values
    const dependencies: string[] = [];
    for (const policy of props.accessPolicies) {
      dependencies.push(policy.fqdn);
    }

    // Add dependency on all access policies
    certificateIssuer.addOverride("depends_on", dependencies);
  }
}
