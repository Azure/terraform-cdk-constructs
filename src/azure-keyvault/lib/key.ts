import {
  KeyVaultKey,
  KeyVaultKeyRotationPolicy,
} from "@cdktf/provider-azurerm/lib/key-vault-key";
import { Construct } from "constructs";
import { AccessPolicy } from "./policy";
import { Vault } from "./vault";

export interface KeyProps {
  /**
   * The name of the key in the Azure Key Vault.
   */
  readonly name: string;

  readonly keyVaultId: Vault;

  /**
   * The type of key to create (e.g., RSA, EC, etc.).
   */
  readonly keyType: string;

  /**
   * The size of the key, typically specified for RSA keys.
   */
  readonly keySize?: number;

  /**
   * Additional options or attributes related to the key.
   */
  readonly keyOpts: string[]; // This is a guess based on the name; adjust the type if needed.

  /**
   * The policy for key rotation.
   */
  readonly rotationPolicy?: KeyVaultKeyRotationPolicy;

  /**
   * Expiration date of the key. Format: UTC, YYYY-MM-DDTHH:MM:SSZ.
   */
  readonly expires?: string;

  readonly accessPolicies: AccessPolicy[];
}

export class Key extends Construct {
  constructor(scope: Construct, id: string, props: KeyProps) {
    super(scope, id);

    const key = new KeyVaultKey(this, "AzureKeyVaultKey", {
      name: props.name,
      keyVaultId: props.keyVaultId.id,
      keyType: props.keyType,
      keySize: props.keySize,
      keyOpts: props.keyOpts,
      rotationPolicy: props.rotationPolicy,
      expirationDate: props.expires,
    });

    // Accumulate all the fqdn values
    const dependencies: string[] = [];
    for (const policy of props.accessPolicies) {
      dependencies.push(policy.fqdn);
    }

    // Add dependency on all access policies
    key.addOverride("depends_on", dependencies);
  }
}
