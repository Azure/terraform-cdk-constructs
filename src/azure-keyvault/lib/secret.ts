import { KeyVaultSecret } from "@cdktf/provider-azurerm/lib/key-vault-secret";
import { Construct } from "constructs";
import { AccessPolicy } from "./policy";
import { Vault } from "./vault";

export interface SecretProps {
  keyVaultId: Vault;
  name: string;
  value: string;
  expirationDate?: string;
  contentType?: string;
  accessPolicies: AccessPolicy[];
}

export class Secret extends Construct {
  constructor(scope: Construct, id: string, props: SecretProps) {
    super(scope, id);

    // Logic to add the secret to the provided keyVault instance
    // For example:
    const secret = new KeyVaultSecret(this, props.name, {
      keyVaultId: props.keyVaultId.id,
      name: props.name,
      value: props.value,
      contentType: props.contentType,
      expirationDate: props.expirationDate,
    });

    // Accumulate all the fqdn values
    const dependencies: string[] = [];
    for (const policy of props.accessPolicies) {
      dependencies.push(policy.fqdn);
    }

    // Add dependency on all access policies
    secret.addOverride("depends_on", dependencies);
  }
}
