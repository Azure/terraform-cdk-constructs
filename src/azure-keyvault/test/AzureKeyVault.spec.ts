import { AzurermProvider } from "@cdktf/provider-azurerm/lib/provider";
import { ResourceGroup } from "@cdktf/provider-azurerm/lib/resource-group";
import { Testing, TerraformStack } from "cdktf";
import { exampleAzureKeyVault } from "./ExampleAzureKeyVault";
import "cdktf/lib/testing/adapters/jest";
import * as kv from "..";
import * as util from "../../util/azureTenantIdHelpers";

describe("Azure Key Vault With Defaults", () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", { features: {} });

    const rg = new ResourceGroup(stack, "MyResourceGroup", {
      name: "rg-test",
      location: "eastus",
    });

    new kv.Vault(stack, "testAzureKeyVaultDefaults", {
      name: "kv-test",
      location: "eastus",
      resourceGroup: rg,
      tenantId: "123e4567-e89b-12d3-a456-426614174000",
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Key Vault with defaults and checks snapshot", () => {
    expect(Testing.synth(stack)).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});

describe("Azure Key Vault Example", () => {
  //let stack: TerraformStack;

  // beforeEach(() => {
  //   const app = Testing.app();
  //   stack = new TerraformStack(app, "test");

  //   Testing.synthScope((stack) => {
  //     new exampleAzureKeyVault(stack, "testAzureKeyVault");
  //   });
  // });

  it("renders the Example Azure KeyVault and checks snapshot", () => {
    // Need to remove the tenant_id from the snapshot as it will change wherever the test is run
    const output = Testing.synth(
      new exampleAzureKeyVault(Testing.app(), "testAzureKeyVault"),
    );
    const myObject: Record<string, any> = JSON.parse(output);

    expect(util.removeTenantIdFromSnapshot(myObject)).toMatchSnapshot();
  });

  it("check if the produced terraform configuration is valid", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(
      Testing.fullSynth(
        new exampleAzureKeyVault(Testing.app(), "testAzureKeyVault"),
      ),
    ).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(
      Testing.fullSynth(
        new exampleAzureKeyVault(Testing.app(), "testAzureKeyVault"),
      ),
    ).toPlanSuccessfully();
  });
});
