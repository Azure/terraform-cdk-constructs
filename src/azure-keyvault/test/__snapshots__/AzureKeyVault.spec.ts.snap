// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Key Vault Example renders the Example Azure KeyVault and checks snapshot 1`] = `
{
  "data": {
    "azurerm_client_config": {
      "CurrentClientConfig": {},
    },
    "azurerm_monitor_diagnostic_categories": {
      "kv_diagsettings_diagcategories_3468E7DD": {
        "resource_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
      },
    },
  },
  "output": {
    "id": {
      "value": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
    },
    "key_vault_name": {
      "value": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.name}",
    },
    "resource_group_name": {
      "value": "\${azurerm_resource_group.rg.name}",
    },
  },
  "provider": {
    "azurerm": [
      {
        "features": {},
      },
    ],
  },
  "resource": {
    "azurerm_key_vault": {
      "kv_key_vault_6CCDEEF8": {
        "location": "eastus",
        "name": "kv-\${var.name}",
        "network_acls": {
          "bypass": "AzureServices",
          "default_action": "Allow",
          "ip_rules": [
            "0.0.0.0/0",
          ],
        },
        "purge_protection_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "sku_name": "standard",
        "soft_delete_retention_days": 7,
      },
    },
    "azurerm_key_vault_access_policy": {
      "kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949": {
        "certificate_permissions": [
          "Get",
          "List",
          "Create",
          "Delete",
          "GetIssuers",
          "ManageIssuers",
        ],
        "key_permissions": [
          "Backup",
          "Create",
          "Decrypt",
          "Delete",
          "Encrypt",
          "Get",
          "Import",
          "List",
          "Purge",
          "Recover",
          "Restore",
          "Sign",
          "UnwrapKey",
          "Update",
          "Verify",
          "WrapKey",
          "Release",
          "Rotate",
          "GetRotationPolicy",
          "SetRotationPolicy",
        ],
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "object_id": "\${data.azurerm_client_config.CurrentClientConfig.object_id}",
        "secret_permissions": [
          "Get",
          "List",
          "Set",
          "Delete",
        ],
        "storage_permissions": [
          "Get",
          "List",
          "Set",
          "Delete",
        ],
      },
      "kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5": {
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "object_id": "bc26a701-6acb-4117-93e0-e44054e22d60",
        "secret_permissions": [
          "Get",
          "List",
          "Set",
          "Delete",
          "Backup",
          "Restore",
          "Recover",
        ],
      },
    },
    "azurerm_key_vault_certificate": {
      "kv_cert1_AzureKeyVaultCertificate_A017BD2B": {
        "certificate_policy": {
          "issuer_parameters": {
            "name": "Self",
          },
          "key_properties": {
            "exportable": true,
            "key_size": 2048,
            "key_type": "RSA",
            "reuse_key": true,
          },
          "lifetime_action": [
            {
              "action": {
                "action_type": "AutoRenew",
              },
              "trigger": {
                "days_before_expiry": 45,
              },
            },
          ],
          "secret_properties": {
            "content_type": "application/x-pkcs12",
          },
          "x509_certificate_properties": {
            "extended_key_usage": [
              "1.3.6.1.5.5.7.3.1",
            ],
            "key_usage": [
              "cRLSign",
              "dataEncipherment",
              "digitalSignature",
              "keyAgreement",
              "keyCertSign",
              "keyEncipherment",
            ],
            "subject": "CN=contoso.com",
            "subject_alternative_names": {
              "dns_names": [
                "internal.contoso.com",
                "domain.hello.world",
              ],
            },
            "validity_in_months": 12,
          },
        },
        "depends_on": [
          "azurerm_key_vault_access_policy.kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5",
          "azurerm_key_vault_access_policy.kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949",
        ],
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "name": "cert1",
      },
    },
    "azurerm_key_vault_certificate_issuer": {
      "kv_issuer1_AzureKeyVaultCertificate_90383EBD": {
        "depends_on": [
          "azurerm_key_vault_access_policy.kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5",
          "azurerm_key_vault_access_policy.kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949",
        ],
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "name": "issuer1",
        "provider_name": "SslAdminV2",
      },
    },
    "azurerm_key_vault_key": {
      "kv_key1_AzureKeyVaultKey_02CD8333": {
        "depends_on": [
          "azurerm_key_vault_access_policy.kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5",
          "azurerm_key_vault_access_policy.kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949",
        ],
        "expiration_date": "2033-08-23T15:23:17Z",
        "key_opts": [
          "decrypt",
          "encrypt",
          "sign",
          "unwrapKey",
          "verify",
          "wrapKey",
        ],
        "key_size": 2048,
        "key_type": "RSA",
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "name": "key1",
      },
      "kv_key2_AzureKeyVaultKey_AB98C26C": {
        "depends_on": [
          "azurerm_key_vault_access_policy.kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5",
          "azurerm_key_vault_access_policy.kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949",
        ],
        "expiration_date": "2033-08-23T15:23:17Z",
        "key_opts": [
          "encrypt",
          "decrypt",
          "sign",
          "verify",
          "wrapKey",
          "unwrapKey",
        ],
        "key_size": 2048,
        "key_type": "RSA",
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "name": "key2",
      },
    },
    "azurerm_key_vault_secret": {
      "kv_customSecretName_69DB8BBD": {
        "content_type": "application/x-pkcs12",
        "depends_on": [
          "azurerm_key_vault_access_policy.kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5",
          "azurerm_key_vault_access_policy.kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949",
        ],
        "expiration_date": "2033-08-23T15:23:17Z",
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "name": "customSecretName",
        "value": "password",
      },
      "kv_secret1_8BCF2182": {
        "content_type": "application/x-pkcs12",
        "depends_on": [
          "azurerm_key_vault_access_policy.kv_kv_secret_admin_access_bc26a701-6acb-4117-93e0-e44054e22d60_policy_BF8031B5",
          "azurerm_key_vault_access_policy.kv_kv_custom_policy_access_TfTokenTOKEN42_policy_3E859949",
        ],
        "expiration_date": "2033-08-23T15:23:17Z",
        "key_vault_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
        "name": "secret1",
        "value": "password",
      },
    },
    "azurerm_log_analytics_workspace": {
      "log_analytics": {
        "location": "eastus",
        "name": "la-\${var.name}",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
      },
    },
    "azurerm_monitor_diagnostic_setting": {
      "kv_diagsettings_CA534DF2": {
        "dynamic": {
          "enabled_log": {
            "content": {
              "category": "\${enabled_log.value}",
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.kv_diagsettings_diagcategories_3468E7DD.log_category_types)}",
          },
          "metric": {
            "content": {
              "category": "\${metric.value}",
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.kv_diagsettings_diagcategories_3468E7DD.metrics)}",
          },
        },
        "log_analytics_workspace_id": "\${azurerm_log_analytics_workspace.log_analytics.id}",
        "name": "diagsettings",
        "target_resource_id": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
      },
    },
    "azurerm_resource_group": {
      "rg": {
        "location": "eastus",
        "name": "rg-\${var.name}",
      },
    },
    "azurerm_role_assignment": {
      "kv_TfTokenTOKEN41Contributor_role_364F22CA": {
        "principal_id": "\${data.azurerm_client_config.CurrentClientConfig.object_id}",
        "role_definition_name": "Contributor",
        "scope": "\${azurerm_key_vault.kv_key_vault_6CCDEEF8.id}",
      },
    },
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.70.0",
      },
    },
  },
  "variable": {
    "name": {
      "default": "test",
      "description": "System name used to randomize the resources",
      "type": "string",
    },
  },
}
`;

exports[`Azure Key Vault With Defaults renders an Azure Key Vault with defaults and checks snapshot 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_key_vault.testAzureKeyVaultDefaults_key_vault_A0C09326.id}"
    },
    "key_vault_name": {
      "value": "\${azurerm_key_vault.testAzureKeyVaultDefaults_key_vault_A0C09326.name}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_key_vault": {
      "testAzureKeyVaultDefaults_key_vault_A0C09326": {
        "location": "eastus",
        "name": "kv-test",
        "purge_protection_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.MyResourceGroup.name}",
        "sku_name": "standard",
        "soft_delete_retention_days": 90,
        "tenant_id": "123e4567-e89b-12d3-a456-426614174000"
      }
    },
    "azurerm_resource_group": {
      "MyResourceGroup": {
        "location": "eastus",
        "name": "rg-test"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.70.0"
      }
    }
  }
}"
`;
